clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/containers/KubernetesCluster.java

```diff
@@ -0,0 +1,152 @@
+        .withCommand(
+            "server",
+            "--kubelet-arg=eviction-hard=imagefs.available<1%,nodefs.available<1%",
+            "--kubelet-arg=eviction-minimum-reclaim=imagefs.available=1%,nodefs.available=1%",
+            "--tls-san",
+            "0.0.0.0")
+        .waitingFor(Wait.forLogMessage(".*Wrote kubeconfig .*", 1));
+  }
+
+  public Path getKubecfgPath() {
+    return kubecfgPath;
+  }
+
+  public String execKubectl(String args) throws IOException, InterruptedException {
+    ProcessBuilder builder = new ProcessBuilder();
+    List<String> cmd = new ArrayList<>();
+    cmd.add("sh");
+    cmd.add("-c");
+    cmd.add(
+        "${PROJECT_ROOT}/clouddriver-kubernetes/src/integration/resources/kubectl-wrapper.sh --kubeconfig="
+            + kubecfgPath
+            + " "
+            + args);
+    builder.command(cmd);
+    builder.redirectErrorStream(true);
+    Process process = builder.start();
+    Reader reader = new InputStreamReader(process.getInputStream(), UTF_8);
+    String output = FileCopyUtils.copyToString(reader);
+    if (!process.waitFor(1, TimeUnit.MINUTES)) {
+      fail("Command %s did not return after one minute", cmd);
+    }
+    assertThat(process.exitValue())
+        .as("Running %s returned non-zero exit code. Output:\n%s", cmd, output)
+        .isEqualTo(0);
+    System.out.println("kubectl " + args + ":\n" + output);
+    return output.trim();
+  }
+
+  @Override
+  public void start() {
+    super.start();
+    String containerName = getContainerInfo().getName().replaceAll("/", "");
+    System.setProperty(this.accountName + "_containername", containerName);
+    try {
+      this.kubecfgPath = copyKubecfgFromCluster(containerName);
+      fixKubeEndpoint(this.kubecfgPath);
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to initialize kubectl or kubeconfig.yml files", e);
+    }
+  }
+
+  @Override
+  public void stop() {
+    super.stop();
+    try {
+      Files.deleteIfExists(this.kubecfgPath);
+    } catch (IOException e) {
+      /* ignored */
+    }
+  }
+
+  private Path copyKubecfgFromCluster(String containerName) throws IOException {
+    Path myKubeconfig =
+        Paths.get(
+            System.getenv("PROJECT_ROOT"),
+            "clouddriver-kubernetes",
+            "build",
+            "kubeconfigs",
+            "kubecfg-" + containerName + ".yml");
+    Files.createDirectories(myKubeconfig.getParent());
+    copyFileFromContainer(KUBECFG_IN_CONTAINER, myKubeconfig.toAbsolutePath().toString());
+    return myKubeconfig;
+  }
+
+  private void fixKubeEndpoint(Path kubecfgPath) throws IOException {
+    String kubeEndpoint = "https://" + getHost() + ":" + getMappedPort(6443);
+    Yaml yaml = new Yaml();
+    InputStream inputStream = Files.newInputStream(kubecfgPath);
+    Map<String, Object> obj = yaml.load(inputStream);
+    List<Map<String, Map<String, String>>> clusters =
+        (List<Map<String, Map<String, String>>>) obj.get("clusters");
```

Comment:
Agree that the unchecked cast here makes sense, but we should add a `@SuppressWarnings("unchecked")` to the method with a brief comment so that we don't get a warning.

You *could* do `KubeConfig obj = KubeConfig.loadKubeConfig(new FileReader(kubecfgPath.toFile()));` but it looks like the `KubeConfig` fields are just `Object` anyway so you'd still need to do the unchecked cast (and suppress it).